src/test/libslic3r/test_arc_fitting.cpp

```diff
@@ -0,0 +1,328 @@
+#include <catch.hpp>
+#include "libslic3r.h"
+#include "Point.hpp"
+#include "Geometry.hpp"
+#include "GCode/ArcFitting.hpp"
+#include "Log.hpp"
+#include "test_data.hpp"
+using namespace Slic3r;
+using namespace Slic3r::Test;
+
+void make_circle(Points &output, size_t n, double r, Point center = Point(0,0), double a = 0.0, double b = 2*PI, double noise = 0.0){
+    double arcdiff = (b-a)/n;
+    for(auto i = 0U; i < n; i++){
+        output.push_back(Point(r*cos(a)+center.x,r*sin(a))+center.y);
+        a += arcdiff;
+    }
+}
+
+SCENARIO("Arc detection works"){
+    GIVEN("A simple line"){
+        Polyline pl;
+        pl.points = {Point(5,15),Point(10,15),Point(15,15), Point(20,15), Point(25,15)};
+        THEN("No arcs are detected"){
+            auto arcs = arc_detection(pl, 4, 1.0);
+            REQUIRE(arcs.size() == pl.points.size());
+            for(auto &arc : arcs){
+                REQUIRE(arc == 0);
+            }
+        }
+    }
+    GIVEN("An arc with too few points"){
+        Polyline pl;
+        make_circle(pl.points,4,9000);
+        THEN("No arcs are detected"){
+            auto arcs = arc_detection(pl, 4, 2.0);
+            REQUIRE(arcs.size() == pl.points.size());
+            for(auto &arc : arcs){
+                REQUIRE(arc == 0);
+            }
+        }
+    }
+    GIVEN("A circle"){
+        Polyline pl;
+        make_circle(pl.points,20,9000, Point(100,100));
+        auto arcs = arc_detection(pl, 4, 10.0);
+        THEN("Output is valid"){
+            REQUIRE(arcs.size() == pl.points.size());
+        }
+        THEN("The last N points don't detect an arc"){
+            for(auto i = arcs.size()-4; i < arcs.size();i++){
+                REQUIRE(arcs[i] == 0);
+            }
+        }
+        THEN("All other points detect an arc"){
+            for(auto i = 0U; i < arcs.size()-4;i++){
+                INFO(i);
+                REQUIRE(arcs.at(i) != 0);
+            }
+        }
+     }
+    GIVEN("A zig zag line"){
+        Polyline pl;
+        pl.points = {Point(0,0),Point(10,10),
+                     Point(20,0),Point(30,10),
+                     Point(40,0),Point(50,10),
+                     Point(60,0),Point(70,10),
+                     Point(80,0),Point(90,10),
+                     Point(100,0),Point(110,10)};
+        THEN("No arcs are detected"){
+            auto arcs = arc_detection(pl,4,1.0);
+            REQUIRE(arcs.size() == pl.points.size());
+            for(auto &arc : arcs){
+                REQUIRE(arc == 0);
+            }
+         }
+    }
+    GIVEN("A line followed by an arc followed by a line"){
+        Polyline pl;
+        pl.points = {Point(-100,900),Point(-70,900),Point(-50,900),Point(-20,900)};
```

Comment:
Again, highly suggest using constants that relate to something that you can reason about, unless you're interested in tiny distances realspace.